spec/models/statement_spec.rb

```diff
@@ -69,6 +69,23 @@
         expect(Statement.with_state("foo", "bar").to_sql).to include(%(WHERE "statements"."state" IN ('foo', 'bar')))
       end
     end
+
+    describe ".with_output_fee" do
+      it "selects only output fee statements" do
+        expect(Statement.with_output_fee.to_sql).to include(%(WHERE "statements"."output_fee" = TRUE))
+      end
+    end
+
+    describe ".next_output_fee_statements" do
+      it "selects the output fee statement with the earliest deadline date in the future" do
+        freeze_time do
+          sql = Statement.next_output_fee_statements.to_sql
```

Comment:
I wonder if we can test those with actual statements like the test in the lead provider spec, mostly because I feel like those can be tested easily without asserting against the SQL?

Reply:
[Updated](https://github.com/DFE-Digital/npq-registration/pull/1460/commits/355e9912dc381169d481721dab2d2afaad23b61a)